app/src/editor/canvas/components/ModuleRenderer/useModuleApi.js

```diff
@@ -0,0 +1,26 @@
+// @flow
+
+import { createContext, useContext, type Context } from 'react'
+
+type ModuleApi = {
+    moduleSidebarOpen: (boolean) => void,
+    port: {
+        onChange: (any, any) => void,
+        setPortOptions: () => void,
+    },
+    selectModule: (any) => void,
+    setCanvas: (any, (any) => void, ?() => void) => void,
+    [string]: () => void,
+}
+
+export const ModuleApiContext = (createContext({
+    moduleSidebarOpen: () => {},
+    port: {
+        onChange: () => {},
+        setPortOptions: () => {},
+    },
+    selectModule: () => {},
+    setCanvas: () => {},
+}): Context<ModuleApi>)
+
+export default (): ModuleApi => useContext(ModuleApiContext)
```

Comment:
The original idea was actually to pass `api` as a prop until we get multi-context with hooks, so this basically matches the plan.